draft-bormann-cbor-yang-standin.md

```diff
@@ -382,18 +354,84 @@ between the producer and the consumer of YANG-CBOR information:
   A consumer that has requirements for only receiving stand-in tags in
   place of legacy representations, MUST indicate this to the producer.
 
-ISSUE: Where do we put those two aspects of negotiation?
+## Intermediate transcoding
 
-* NETCONF negotiation
-* yang-library
-* media-type parameters
-* ?
+The simplest situation is when no intermediate encoders and decoders are
+involved in the data transfer, therefore the round trip is not legacy.
+In this case, no conversions are involved.
+
+Producing a stand-in MUST be always triggered by schema usage. Intermediate encoders
+MUST NOT encode stand-ins when no schema is available. This avoids problems with
```

Comment:
This paragraph is explicitly targeted on intermediate encoders without schema. Let's say the intermediate encoder chooses hex string over date, and encodes `15:32:46` as `43 15 32 46`. But the schema (which the intermediate encoder isn't aware of) says `time`. This looks quite confusing for me when expecting either `68 31 35 3A 33 32 3A 34 36` or … aha, we don't have a tag for just a time not bound to a date, probably. That looks like an issue.

Anyway, what I wanted to say, is that one string may actually have many different meanings depending on context, which is what we are already experiencing in BIRD configuration language. With that, I'm advocating for "standins strictly only with schema", otherwise we're getting into a rabbit hole of reinterpreting e.g. times to hex strings, or even some weirder things with e.g. BGP large communities … which can be also `15:32:46`, see draft-ietf-idr-bgp-model-17 and `bgp-large-community-type` … but this time encoded probably as something like `4C 00 00 00 0F 00 00 00 20 00 00 00 2E` probably with an appropriate tag. And that's what the implementors of CBOR decoders are going to hate us for if we allow it.

Reply:
Do you have an example for a rfc 6991 "time"?
I wasn't planning to spend time on this; tag 1001 of course can model this as the duration from a start-of-day (which is even more compact than a hex0string :-).